src/input_xml.F90

```diff
@@ -5628,17 +4632,206 @@ subroutine generate_rpn(cell_id, tokens, output)
   end subroutine generate_rpn
 
 !===============================================================================
-! NORMALIZE_AO normalizes the atom or weight percentages for each material
+! ASSIGN_NUCLIDE_DENSITIES Read in the nuclide densities from materials.xml
+! and assign to the corresponding nuclides.
+!===============================================================================
+
+  subroutine assign_nuclide_densities()
```

Comment:
Looks like that was something that just needed to be done when performing that natural element expansion in fortran for elements input as weight percents. I'll combine this into `read_materials_xml`.